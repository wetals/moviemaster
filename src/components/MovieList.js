import React from "react";
import Movie from "./Movie";

const MovieList = props => {
  const { genreIds, rating } = props.visibilityFilter;

  const isGenreVisible = movieIds => {
    if (genreIds.length === 0) return true;
    return genreIds.every(
      currentValue => movieIds.indexOf(Number(currentValue)) > -1
    );
  };

  const filteredMovies = props.movies.filter(
    movie => isGenreVisible(movie.genre_ids) && movie.vote_average >= rating
  );

  return (
    <div className="items">
      {filteredMovies.length > 0 &&
        filteredMovies.map((movie, key) => {
          return (
            <Movie
              movie={movie}
              images={props.images}
              genresList={props.genresList}
              key={key}
            />
          );
        })}
      {filteredMovies.length === 0 && <div>No results found</div>}
    </div>
  );
};

export default MovieList;
