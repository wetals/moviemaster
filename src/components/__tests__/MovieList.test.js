import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import MovieList from "../MovieList";

const testData = {
  movies: [
    {
      title: "foo",
      genre_ids: [
        {
          id: 1,
          name: "one"
        }
      ],
      poster_path: "1.jpg",
      vote_average: 5
    },
    {
      title: "bar",
      genre_ids: [
        {
          id: 2,
          name: "two"
        }
      ],
      poster_path: "2.jpg",
      vote_average: 6
    }
  ],
  images: {
    base_url: "url",
    poster_sizes: ["size-1", "size-2", "size-3", "size-4", "size-5"]
  },
  genresList: {
    1: "foo"
  },
  visibilityFilter: {
    genreIds: [],
    rating: 3
  }
};

describe("MovieList component", () => {
  it("renders properly", () => {
    const wrapper = shallow(<MovieList {...testData} />);
    const element = wrapper.find(".items");
    expect(element.children()).toHaveLength(testData.movies.length);
  });

  it("renders no item when ratings is below filter selection", () => {
    const testProps = {
      ...testData,
      visibilityFilter: {
        genreIds: [],
        rating: 9
      }
    };
    const wrapper = shallow(<MovieList {...testProps} />);
    const element = wrapper.find(".items");
    expect(element.text()).toContain("No results");
  });

  it("renders the right number of items when filters are applied", () => {
    const testProps = {
      ...testData,
      visibilityFilter: {
        genreIds: ["2"],
        rating: 9
      }
    };
    const wrapper = shallow(<MovieList {...testProps} />);
    const element = wrapper.find(".items");
    expect(element.children()).toHaveLength(1);
  });

  it("renders snapshot properly", () => {
    const tree = renderer.create(<MovieList {...testData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
