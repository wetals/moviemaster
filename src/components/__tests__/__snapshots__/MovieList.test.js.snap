// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component renders snapshot properly 1`] = `
<div
  className="items"
>
  <article
    className="item"
  >
    <div
      className="item__image"
    >
      <img
        alt="foo"
        src="urlsize-41.jpg"
      />
    </div>
    <div
      className="item__metadata"
    >
      <div
        className="item__title"
      >
        foo
      </div>
      <div
        className="genre__tags"
      >
        <span
          className="genre"
        />
      </div>
    </div>
  </article>
  <article
    className="item"
  >
    <div
      className="item__image"
    >
      <img
        alt="bar"
        src="urlsize-42.jpg"
      />
    </div>
    <div
      className="item__metadata"
    >
      <div
        className="item__title"
      >
        bar
      </div>
      <div
        className="genre__tags"
      >
        <span
          className="genre"
        />
      </div>
    </div>
  </article>
</div>
`;
